heartbeat/monitors/task.go

```diff
@@ -100,7 +104,7 @@ func (t *configuredJob) Stop() {
 	}
 }
 
-func runPublishJob(job jobs.Job, client beat.Client) []scheduler.TaskFunc {
+func runPublishJob(job jobs.Job, client *core.SyncClient) []scheduler.TaskFunc {
```

Comment:
Does this need to explicitly be a `SyncClient` or does `beat.Client` work? Should we use `SyncClient` everywhere, or do we want to only use it in `RunOnce` mode?

Reply:
`beat.Client` does not have a `Wait` function, so no. I think you should only use it when you want to wait until the publishing has finished. In other Beats it does not make sense. Why block in a function when you can achieve something asynchronously? :)